File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int l = 0;
        int r = height.length-1;
        int max = 0;
        while(l<r){
            int left_height = height[l];
            int right_height = height[r];
            int min_height = Math.min(left_height, right_height);
            max = Math.max(max, min_height * (r - l)); 
            if(left_height < right_height) l++;
            else r--;
        }
        return max;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int l = 0;
        int r = height.size()-1;
        int maxArea = 0;
        while(l<r){
            int left_height = height[l];
            int right_height = height[r];
            int min_height = min(left_height, right_height);
            maxArea = max(maxArea, min_height * (r - l)); 
            if(left_height < right_height){ 
                l++;
            }
            else {
                r--;
            }
        }
        return maxArea;
    }
};
        
